torch/csrc/api/src/optim/optimizer.cpp

```diff
@@ -75,17 +75,17 @@ void OptimizerOptions::serialize(torch::serialize::OutputArchive& archive) const
     "You must override it in your subclass of torch::optim::OptimizerCloneableOptions<YourOptimizerOptions>.");
 }
 
-namespace detail {
-OptimizerBase::OptimizerBase(std::vector<Tensor> parameters)
-    : parameters_(std::move(parameters)) {}
-
-void OptimizerBase::add_param_group(const OptimizerParamGroup& param_group) {
+void Optimizer::add_param_group(const OptimizerParamGroup& param_group) {
   for (const auto& param : param_group.params()) {
     TORCH_CHECK(param.is_leaf(), "can't optimize a non-leaf Tensor");
   }
   OptimizerParamGroup param_group_(param_group.params());
   if (!param_group.has_options()) {
-    param_group_.set_options(defaults_->clone());
+    if(defaults_ != nullptr) {
```

Comment:
Since `default_` won't be a `nullptr` now (since we require users to pass in a valid defaults value in the Optimizer constructors), we can probably revert to the old code for this part

Reply:
And it would be good to just add a `TORCH_INTERNAL_ASSERT(defaults_ != nullptr)` here :D